frontend/src/metabase/lib/entity-id/hooks/use-validated-entity-id/use-validated-entity-id.ts

```diff
@@ -101,3 +113,12 @@ export const useValidatedEntityId = <
     return { id: null, isLoading: false, isError: true } as const;
   }, [isEntityId, isLoading, isError, entity_ids, id]);
 };
+
+const isStringifiedNumber = (value: unknown): boolean => {
+  if (value === null || value === undefined) {
+    return false;
+  }
+
+  const numberValue = Number(value);
+  return !isNaN(numberValue) && isFinite(numberValue);
```

Comment:
TIL https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isFinite

It seems `isFinite` already checks for `!isNaN` so we probably don't need to check it again here.

```suggestion
  return isFinite(numberValue);
```